ocaml-eglot-util.el

```diff
@@ -20,6 +20,22 @@
 (require 'eglot)
 (require 'cl-lib)
 
+(defclass ocaml-eglot-server (eglot-lsp-server) ()
+  :documentation "OCaml-eglot Language Server.")
+
+(defun ocaml-eglot-util--make-server (orig-fn &rest args)
+  "Wrap `make-instance` to return `ocaml-eglot-server` using ORIG-FN and ARGS."
+  (let ((class (car args)))
+    (if (and (eq class 'eglot-lsp-server)
+             (member major-mode '(tuareg-mode
+                                  caml-mode
+                                  neocaml-mode
+                                  neocamli-mode)))
+        (apply orig-fn 'ocaml-eglot-server (cdr args))
+      (apply orig-fn args))))
+
+(advice-add 'make-instance :around #'ocaml-eglot-util--make-server)
```

Comment:
Using advice like this is bad, we should figure out some other way to do whatever we're trying to do here.

Reply:
I was not really able to handle my own server so I try to containt advise inside setup/clean function: https://github.com/tarides/ocaml-eglot/pull/64/commits/a900a0dbb911e6f2e6eef5d192ba56530446f0b5 is that better or do you have an other idea?

Reply:
Ah, that's not really an improvement unfortunately, it's still using advice which is definitely wrong.  Why do you need to return ocaml-eglot-server as the server type anyway?  Can you just avoid that?